src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -60,10 +62,13 @@ public class AppiumCommandExecutor extends HttpCommandExecutor {
 
     private final Optional<DriverService> serviceOptional;
 
+    private final AppiumClientConfig appiumClientConfig;
+
     private AppiumCommandExecutor(Map<String, CommandInfo> additionalCommands, DriverService service,
                                   URL addressOfRemoteServer,
                                   HttpClient.Factory httpClientFactory,
-                                  ClientConfig clientConfig) {
+                                  ClientConfig clientConfig,
+                                  AppiumClientConfig appiumClientConfig) {
```

Comment:
Does it make sense to inherit `ClientConfig` in `AppiumClientConfig` and then use only `AppiumClientConfig`?

Old methods with `ClientConfig` can be marked as `@Deprecated`

Reply:
+1.  We have not released a new version post introducing client config so we don't have to deprecate it. We can have AppiumClientConfig inherit ClientConfig and use it across.

Reply:
Perhaps could be. The ClientConfig is in selenium, so I defined only Appium thing separately

Reply:
Btw, how can we inherit private constructor? (ClientConfig)
https://github.com/SeleniumHQ/selenium/blob/9b012df78dbbfd84d8fef06308f40fbf6a29455b/java/src/org/openqa/selenium/remote/http/ClientConfig.java#L42

Maybe I forgot something in Java, or probably haven't tried to extend private constructor's class in my past...

Reply:
I'm afraid it's not possible in general OOP paradigm, reflection might be overhead here, it seems only composition can be used

Reply:
I believe we can ask Selenium fellows to help us making it public. cc @titusfortner @diemol

Reply:
I'm not sure why this design pattern was chosen for this class. Make a PR to change the constructor to protected and we'll get it released.

Reply:
Thanks, I have created https://github.com/SeleniumHQ/selenium/pull/10985